rust/src/delta_datafusion.rs

```diff
@@ -224,6 +257,67 @@ fn to_scalar_value(stat_val: &serde_json::Value) -> Option<datafusion::scalar::S
     }
 }
 
+fn correct_scalar_value_type(
+    value: datafusion::scalar::ScalarValue,
+    field_dt: &ArrowDataType,
+) -> datafusion::scalar::ScalarValue {
+    match field_dt {
+        ArrowDataType::Int64 => {
+            let raw_value = i64::try_from(value).unwrap();
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Int32 => {
+            let raw_value = i64::try_from(value).unwrap() as i32;
```

Comment:
Is there any reason why we prefer downcasting from i64 to i32 over direct `i32::try_from(value)` call?

Reply:
`i32::try_from` call only applies on an input of `ScalarValue::Int32`. For number scalar value, we all use `ScalarValue::Int64` to compare to get max/min values. That's said here the input is `ScalarValue::Int64`.

Reply:
ha, i see. perhaps in the future, we could change `to_scalar_value` to do the scalar value type cast based on field type so we can avoid this intermediate 64 bits conversion overhead.